apps/src/templates/projects/GalleryReportAbusePopUp.jsx

```diff
@@ -0,0 +1,240 @@
+      'authenticity_token',
+      RailsAuthenticityToken.getRailsCSRFMetaTags().token
+    );
+    formData.append('channel_id', this.props.projectData.channel);
+    formData.append('abuse_url', this.props.abuseUrl);
+    formData.append('g-recaptcha-response', this.token);
+
+    const checkedCheckboxNames = this.state.checkboxes
+      .filter(checkbox => checkbox.checked)
+      .map(checkbox => checkbox.name);
+    formData.append('abuse_type', checkedCheckboxNames);
+
+    fetch('/report_abuse_pop_up', {
+      method: 'POST',
+      body: formData,
+    })
+      .then(response => {
+        if (response.ok) {
+          this.setState({
+            showSubmitConfirmation: true,
+          });
+          this.props.onReport();
+        } else if (response.status === 403) {
+          this.onCaptchaExpiration();
+        }
+      })
+      .catch(error => {
+        this.setState({
+          submitButtonEnabled: false,
+        });
+        console.error('Error sending report data:', error);
+      });
+  }
+
+  onCaptchaExpiration() {
+    this.setState({
+      submitButtonEnabled: false,
+      captchaCompleted: false,
+    });
+  }
+
+  handleCheckboxChange = checkboxName => {
+    this.setState(prevState => {
+      const updatedCheckboxes = prevState.checkboxes.map(checkbox =>
+        checkbox.name === checkboxName
+          ? {...checkbox, checked: !checkbox.checked}
+          : checkbox
+      );
+
+      const isAnyCheckboxSelected = updatedCheckboxes.some(
+        checkbox => checkbox.checked
+      );
+
+      return {
+        checkboxes: updatedCheckboxes,
+        isAnyCheckboxSelected: isAnyCheckboxSelected,
+      };
+    });
+  };
+
+  render() {
+    const {
+      isAnyCheckboxSelected,
+      checkboxes,
+      submitButtonEnabled,
+      showSubmitConfirmation,
+    } = this.state;
+
+    const captchaSiteKey = this.props.recaptchaSiteKey;
+
+    const script = document.createElement('script');
+    script.src = 'https://www.google.com/recaptcha/api.js';
+    script.id = 'captcha';
+    script.async = true;
+    script.defer = true;
+    window.onCaptchaSubmit = token => this.onCaptchaVerification(token);
+    window.onCaptchaExpired = () => this.onCaptchaExpiration();
+    script.onload = () => this.setState({loadedCaptcha: true});
+    document.body.appendChild(script);
+
+    return (
+      <AccessibleDialog
+        className={style.reportAbusePopUp}
+        onClose={this.cancel}
+      >
+        {showSubmitConfirmation ? (
+          <div className={style.submitConfirmation}>
+            <h3>{i18n.thankyou()}!</h3>
+            <p>{i18n.thankYouForReport()}</p>
+            <Button
+              onClick={this.cancel}
+              text={'Continue'}
+              color={Button.ButtonColor.brandSecondaryDefault}
+            />
+          </div>
+        ) : (
+          <div>
+            <div className={style.popUpTitle}>
+              <h3 style={{margin: 0}}>{i18n.reportAbuse()}</h3>
```

Comment:
Shouldn't this h3 styled as componentsLibrary(Typography) `apps/src/componentLibrary/typography/index.ts` `Heading3`?
Also shouldn't the cancel button styled as our `Button` component you've used in other places?

Reply:
@AfifahK oh, now that I see there are multiple `h3`s and `p` - shouldn't all of them be styled as our design system typography components? (if yes - then you'll need to use elements from `apps/src/componentLibrary/typography/index.ts` )

Do you have a figma deisgns link maybe so we can check if we need to use existing typography or not?

Reply:
test